### PR TITLE
Fix #20017 - Avoid ugly initial flash upon load

### LINKED ISSUE #20017: Replace fox head loading screen when user is logged in
## Problem
When you open up the extension popup, there's small delay and for half a second the fox head loading screen appears before the wallet screen loads. The visual flicker that makes it feel a little jittery. We should investigate why this is happening. 

Reference: [Slack thread](https://consensys.slack.com/archives/GTQAGKY5V/p1689289026081309)
Reproduced screencap: https://images.zenhubusercontent.com/42009758/88806063-b72f-4985-8a9a-b7a22c2094dc/screen_recording_2023_07_13_at_7_07_47_pm.mov

## Possible solution
We could possibly introduce a different loading screen (faded, delayed, transparent etc.) when the user's already logged in.

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

* Fixes #20017

This is an initial POC into how we can avoid the ugly flash at the beginning of initialization in the extension.  Presently, with this PR, we can avoid the flash by doing two things:

1.  Initializing the MetaMask background to have the background color that's the system preference (light vs. dark)
2. Delaying initial visibility of the MetaMask logo for a period of time so that it doesn't jarringly display and then not.

Further improvements (?):

1.  Always show the MetaMask logo, fade it out, and fade in the UI 
2. Store the user preference to `localStorage` so that a background color flash doesn't happen if they're using MM in light mode on dark preference machine, and visa versa

## Long Load

https://github.com/MetaMask/metamask-extension/assets/46655/6340a882-9e3e-4542-a730-af4cb776a5bb

## Normal Quick Load

https://github.com/MetaMask/metamask-extension/assets/46655/6f11eb38-e975-4f30-b047-45c59424decc

## Manual Testing Steps

Coming soon

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/css/base-styles.scss [hunk 1]
```diff
@@ -47,10 +47,14 @@ html {
   height: 100%;
   display: flex;
   flex-direction: column;
-  background: var(--color-background-alternative);
 
-  @include screen-sm-max {
-    background-color: var(--color-background-default);
+  /*
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/css/base-styles.scss:51]: https://github.com/MetaMask/metamask-extension/assets/8112138/4ec001bf-68fe-41a9-abd3-4c0a2ec54290

**Review by georgewrmarshall — COMMENTED** [review]
Love the idea but I'm not a huge fan of using the brand colors because we could get out of sync with the theme level tokens. We are also eliminating some of the power of the design tokens. If we were to change the background-default and alternative color values we would need to remember to adjust them here. But I "think" I understand why we have to do it this way

**Review by NidhiKJha — COMMENTED** [review]
This is a nice idea @darkwing but like @georgewrmarshall mentioned, this is breaking in the expanded view and the background is not getting updated in expanded view as well. So, might need to look into that

**Comment by darkwing** [comment]
Updated fade in, fade out:


https://github.com/MetaMask/metamask-extension/assets/46655/6f1dd708-7a3f-4aa8-8c72-824b239adacf

**Comment by HowardBraham** [comment]
If your computer is on Dark Mode, it works properly.  But if your computer is on Light and only your MetaMask is set to Dark, it still flashes white.  (I tested this on Windows, but it could also be true on other platforms)

**Comment by kevinghim** [comment]
Optional: we should track the frequency of the loading screen appearing and the length of time it appears (in ms).

**Comment by darkwing** [comment]
> If your computer is on Dark Mode, it works properly. But if your computer is on Light and only your MetaMask is set to Dark, it still flashes white. (I tested this on Windows, but it could also be true on other platforms)

So yeah, it's a tough spot when the user has chosen a theme opposite of their system theme, since their preference is stored in state (which we don't immediately have), and our best guess is that the user uses system theme (since that's been the default for ~1 year).

We *could* store the user's preference in localStorage, but that poses a few problems:

1.  They'd still see that flash at least one more time (since localStorage would initialize as empty)
2.  Not sure yet what we'd store -- the CSS color, the theme and then hope the theme color doesn't change?

Lots to think about here.

File: ui/css/base-styles.scss [hunk 2]
```diff
@@ -17,6 +17,30 @@ body {
...
+  }
+
+  /*
+    Until we get the user's preference from React state,
+    assume the user is in dark mode, because the default
+    theme is system preference
+  */
+  @media (prefers-color-scheme: dark) {
+    &:not([data-theme]) {
+      background-color: var(--brand-colors-grey-grey900);
```
Comment (georgewrmarshall) [inline ui/css/base-styles.scss:35]: We may have to include color here too to fix the errors page
```suggestion
      color: var(--brand-colors-white-white000);
      background-color: var(--brand-colors-grey-grey900);
```

Screenshot 2023-08-16 at 9 36 43 PM

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! I found a couple issues one of them the same as @HowardBraham with dark loading screen on light. It would be great to get some manual testing steps as wel


https://github.com/MetaMask/metamask-extension/assets/8112138/c89e0264-6840-42f1-9267-7b75b45f87f2

**Review by NidhiKJha — COMMENTED** [review]
Looks good, the only issue I am seeing is the different loading color in the light theme like Howard

File: ui/css/base-styles.scss (hunk 2 above)
Comment (NidhiKJha) [reply ui/css/base-styles.scss:36]: I think we are missing something in this solution, I am still seeing this issue here.
Screenshot 2023-08-18 at 9 37 51 PM

**Comment by HowardBraham** [comment]
It still flashes when the user has chosen a MM theme opposite of their system theme, are we giving up on that?

**Comment by darkwing** [comment]
@kevinghim Do we have numbers on the percentage of users who use a given theme?

File: ui/css/base-styles.scss (hunk 2 above)
Comment (darkwing) [reply ui/css/base-styles.scss:35]: Done!

**Pull request closed without merging**